core/state_transition.go

```diff
@@ -220,6 +223,11 @@ func (st *StateTransition) preCheck() error {
 				st.msg.From().Hex(), msgNonce, stNonce)
 		}
 	}
+	// Make sure the sender is an EOA
+	if codeHash := st.state.GetCodeHash(st.msg.From()); codeHash != emptyCodeHash && codeHash != (common.Hash{}) {
+		return fmt.Errorf("%w: address %v, codehash: %s", ErrSenderNoEOA,
```

Comment:
Please use "%w: address %v, codehash: %x"

Reply:
If we use `%x` on `st.mst.From()`, we don't get the checksummed address. IMO it's better the way it is. As for the codehash, `%x` makes no difference, aside from the `0x` prefix being dropped if we do `%x`.

Reply:
Yes right. I was wrong here.